### PR TITLE
[RM Chap 3] RST Updates - Virtual Storage - Tenant Reference (section…

### PR DESCRIPTION
**Pull request by jbhartley**
…: Virtual Storage) & Table 3.9 - Storage Use Cases and Stereotypes

[RM Chap 3] RST Updates 

Virtual Storage - Correct Reference (was self referencing, rather than referencing to Section: "Storage for Tenant Consumption"

Updated Table 3.9 "Storage Use Cases and Stereotypes" - So correctly spans columns using RST "Grid Table" formatting.

### CODE REVIEW

**Review by walterkozlowski — APPROVED** [review]

**Comment by walterkozlowski** [comment]
@jbhartley is this PR finalised?  If yes, we need more reviews!

**Comment by jbhartley** [comment]
@walterkozlowski @TFredberg @CsatariGergely I have done rst bullet list sweep through the entire section.
I ask if:
- Tomas could please review the networking sections to verify that the multiple bulleted lists are all ok (I have done a check against prior and it seems ok, but have author redo check will ensure we have clear rst version going for format
- Csatari if you could give the section a sweep for rst formatting/mark ups being correct

There is no significant material update at the moment, updates are to get into rst and table in storage reformatted in preparation for next release, where can focus more on content updates.

**Comment by pgoyal01** [comment]
A couple of issues (ignoring many spelling typo's):
1. Most specifications of Edge deployment restrict it to user/data plane viz., no control plane.
2. SImilarly, block storage but no object storage.

Are there deployment scenarios where the above is not true? If yes, they should be listed as exceptions.

File: doc/ref_model/chapters/chapter03.rst [hunk 1]
```diff
@@ -5,13 +5,13 @@ It is necessary to clearly define the infrastructure resources and their capabil
...
--  **Simplicity:** the amount of different types of resources (including their attributes and relationships amongst one another) should be kept to a minimum to reduce the configuration spectrum which needs to be considered
--  **Declarative**: the model should allow for the description of the intended state and configuration of the cloud infrastructure resources for automated life cycle management
--  **Explicit:** the model needs to be rich enough to cover the instantiation and the on-going operation of the cloud infrastructure
--  **Lifecycle:** the model must distinguish between resources which have independent lifecycles but should group together those resources which share a common lifecycle
--  **Aligned:** the model should clearly highlight the dependencies between its components to allow for a well-defined and simplified synchronisation of independent automation tasks.
+- **Scope:** the model should describe the most relevant virtualised infrastructure resources (incl. acceleration technologies) a cloud infrastructure needs to host Telco workloads
```
Comment (ulikleber) [inline doc/ref_model/chapters/chapter03.rst:8]: why don't we line wrap to 120 if we touch some lines? (whole contribution)

File: doc/ref_model/chapters/chapter03.rst [hunk 2]
```diff
@@ -279,58 +279,50 @@ Network Principles
...
--  Scalability: Network model enables scalability to handle all traffic traverse North-South and East-West enabling small up to large deployments in a non-blocking manner.
-
--  Workload agnostic: Network model is capable of providing connectivity to any type of workloads, including VNF, CNF and BareMetal workloads.
-
--  Carrier Grade: Network model is capable of supporting deployments of the carrier grade workloads.
-
--  Future proof: Network model is extendible to support known and emerging technology trends including SmartNICs, FPGAs and Programmable Switches, integrated for multi-clouds, and Edge related technologies.
+- Agnosticism: Define Network Fabric concepts and models that can carry any type of traffic in terms of:
+  - Control, User and Management traffic types
+  - Acceleration technologies that can support multiple types of infrastructure deployments and network function workloads
```
Comment (ulikleber) [inline doc/ref_model/chapters/chapter03.rst:290]: Not shown as separate bullets any more.

**Review by ulikleber — COMMENTED** [review]
Only editorial, but shouldn't we wrap to 120 chars per line?

File: doc/ref_model/chapters/chapter03.rst [hunk 3]
```diff
@@ -30,17 +30,17 @@ The functionalities of each layer are as follows:
 
 **Virtual Infrastructure Layer**
 
--  **Virtual infrastructure resources:** These are all the infrastructure resources (compute, storage and networks) which the cloud infrastructure provides to the VNF/CNF and other workloads. These virtual resources can be managed by the tenants and tenant workloads directly or indirectly via an application programming interface (API).
--  **Virtual infrastructure manager:** This consists of the software components that manage the virtual resources and make those management capabilities accessible via one or more APIs. The responsibilities of this functionality include the management of logical constructs such as tenants, tenant workloads, resource catalogues, identities, access controls, security policies, etc.
+- **Virtual infrastructure resources:** These are all the infrastructure resources (compute, storage and networks) which the cloud infrastructure provides to the VNF/CNF and other workloads. These virtual resources can be managed by the tenants and tenant workloads directly or indirectly via an application programming interface (API).
```
Comment (CsatariGergely) [inline doc/ref_model/chapters/chapter03.rst:33]: Use only one space between the - and the content.

File: doc/ref_model/chapters/chapter03.rst [hunk 4]
```diff
@@ -548,35 +540,32 @@ The creation of the SFC might include design/preparation phase as:
...
 1. Creation of service functions of SFC.
 
--  The flow of steps to enable the SFC creation can be as follows:
-
-   a. SFC orchestrator creates the SFs with help of VNF MANO or CNF MANO.
-   b. SFC Renderer attaches the SFC aware interfaces at SFs to enable Service plane
-   c. NFVO boots up the relevant SF configurations at SF.
-
+   - The flow of steps to enable the SFC creation can be as follows:
+     a. SFC orchestrator creates the SFs with help of VNF MANO or CNF MANO.
```
Comment (CsatariGergely) [inline doc/ref_model/chapters/chapter03.rst:544]: Leave an empty line before the a-b-c list.

File: doc/ref_model/chapters/chapter03.rst [hunk 5]
```diff
@@ -787,139 +776,217 @@ Storage Scenarios and Architecture Fit
...
-Small data-centre Storage    Converged Software Defined Storage  O                  O    O   Y                   Y                    O           O     O   O    O
-Edge Cloud                   Edge Cloud for VNF/CNF Storage      NA                 O    NA  Y                   Y                    O           O     O   O    O
-\                            Edge Cloud for Apps Storage         NA                 O    NA  Y                   Y                    O           O     O   O    Y
-\                            Edge Cloud for Content Mgt Storage  NA                 O    NA  Y                   Y                    O           O     O   O    Y
-============================ =================================== ================== ==== === =================== ==================== =========== ===== === ==== ======
+- "Y" - Yes and almost always provided
+- "O" - Optional and readily accommodated
+- "N" - No, not available
+- "NA" - Not Applicable for this Use Case / Stereotype
++------------------------------+-------------------------------------+------+------+------+------------+------------+--------+-------+-----+------+--------+
```
Comment (CsatariGergely) [inline doc/ref_model/chapters/chapter03.rst:790]: Leave an empty line before the table. 

Add + only to where the line intesects with an other one.

File: doc/ref_model/chapters/chapter03.rst [hunk 6]
```diff
@@ -787,139 +776,217 @@ Storage Scenarios and Architecture Fit
...
+|                                                                                         | Tenant / User                                                  |
++==============================+=====================================+======+======+======+============+============+========+=======+=====+======+========+
```
Comment (CsatariGergely) [inline doc/ref_model/chapters/chapter03.rst:792]: Add + only to where the line intesects with an other one.

File: doc/ref_model/chapters/chapter03.rst [hunk 7]
```diff
@@ -787,139 +776,217 @@ Storage Scenarios and Architecture Fit
...
-
-      1. Embedded Shared File Storage:
+The storage sub-system is a foundational part of any Cloud Infrastructure, as such it is important to identify the storage needs, based on target tenant use cases, at inception. This will allow the right set of considerations to be addressed for the deployment. A set of typical considerations is provided:
+
+- for various use cases to meet functional and performance needs and
+- to avoid the need for signifiant rework of the storage solution and the likely ripple through impact on the broader Cloud Infrastructure.
+
+The considerations will help to guide the build and deployment of the Storage solution for the various Use Cases and Stereotypes outlined in the summary table.
+
++----+--------------------+----+------------------------------------------------------------------------------------------------------------------------------------+
```
Comment (CsatariGergely) [inline doc/ref_model/chapters/chapter03.rst:824]: Add + only to where the line intesects with an other one.

File: doc/ref_model/chapters/chapter03.rst [hunk 8]
```diff
@@ -787,139 +776,217 @@ Storage Scenarios and Architecture Fit
...
+| Use Case                | Description                                                                                                                             |
++====+====================+====+====================================================================================================================================+
```
Comment (CsatariGergely) [inline doc/ref_model/chapters/chapter03.rst:826]: Add + only to where the line intesects with an other one.

**Review by CsatariGergely — CHANGES_REQUESTED** [review]

**Review by jbhartley — COMMENTED** [review]
@pgoyal01 thank you for table updates. Adding Bold to the section areas improves readability. I will now do some additions to cover:
Definitions of: Infra. Control (Ctrl), Management (Mgt) for clarity (with examples)
Add Separated Ctrl / Data Plane Edge Deployment case, as discussed

File: doc/ref_model/chapters/chapter03.rst (hunk 1 above)
Comment (jbhartley) [reply doc/ref_model/chapters/chapter03.rst:12]: @ulikleber @CsatariGergely I have gone through document and done manual 120 character wrap. However this breaks rendering of bulleted list and so I have wound manual wrapping of these lists.
I am now complete with 120 wrapping to the extent that is does  not introduce other issues.

File: doc/ref_model/chapters/chapter03.rst (hunk 3 above)
Comment (jbhartley) [reply doc/ref_model/chapters/chapter03.rst:44]: @CsatariGergely I have done sweep to remove these cases.

File: doc/ref_model/chapters/chapter03.rst (hunk 5 above)
Comment (jbhartley) [reply doc/ref_model/chapters/chapter03.rst:977]: @CsatariGergely done , thank you.

File: doc/ref_model/chapters/chapter03.rst (hunk 4 above)
Comment (jbhartley) [reply doc/ref_model/chapters/chapter03.rst:688]: @CsatariGergely done.

File: doc/ref_model/chapters/chapter03.rst (hunk 6 above)
Comment (jbhartley) [reply doc/ref_model/chapters/chapter03.rst:980]: @CsatariGergely tables render correctly and having the markers (+) help with ensuring alignment across merged cells.

**Comment by jbhartley** [comment]
@walterkozlowski @pgoyal01 @TFredberg rst tables and bulleted lists are all complete now. Requesting review to make sure that these are all correct and no document structure has been lost in conversion.

File: doc/ref_model/chapters/chapter03.rst [hunk 9]
```diff
@@ -1178,7 +1178,7 @@ The considerations will help to guide the build and deployment of the Storage so
 |    | can bring with it considerations that are specific to this choice             |
 +----+----+----+----------+----------------------------------------------------------+
 |    | Converged Software Defined Storage                                            |
-|----+----+----+----------+----------------------------------------------------------+
++----+----+----+----------+----------------------------------------------------------+
```
Comment (CsatariGergely) [inline doc/ref_model/chapters/chapter03.rst:1181]:
```suggestion
+---------+----+---------------------------------------------------------------------+
```

Comment (CsatariGergely) [reply doc/ref_model/chapters/chapter03.rst:1181]: It is difficult to edit multiline stuff in the GitHub UI for comments, so I'm, not sure if this is totally correct, but I hope you get the idea.

**Review by CsatariGergely — CHANGES_REQUESTED** [review]

**Review by ulikleber — APPROVED** [review]
Fine from my side.

**Comment by jbhartley** [comment]
> You should align the wrapped lines to the beginning of the text

@CsatariGergely @walterkozlowski can we please merge and commit content updates so we can review top to bottom via "Read The Doc" and then revisit rst formatting.

This will also allow merge and review with new pull request: https://github.com/cntt-n/CNTT/pull/2986 to ensure that any conflicts (if the exists can be resolved)

**Comment by walterkozlowski** [comment]
@jbhartley and @CsatariGergely, can you please agree on how to address the editing issues, so we can go ahead and merge this needed and important PR

**Comment by jbhartley** [comment]
@walterkozlowski @CsatariGergely @pgoyal01 I have resolved all merge conflicts for this pull requests by:
1. Accepted all non-storage updates from master (with corresponding 120 Wrap applied)
2. Merging conflicts within Storage related sections to ensure most current material has been selected.
3. Applying outstanding wrapping to items (2).

**Review by pgoyal01 — APPROVED** [review]

**Comment by walterkozlowski** [comment]
@CsatariGergely, we have an agreement on the contents of this PR ( in RM meeting discussion, and three approvals in this PR), and @jbhartley reports he has fixed the formatting, so this PR needs to be merged. Please approve on Monday at the latest, so we can merge and close Issues.  Thank you.

File: doc/ref_model/chapters/chapter03.rst [hunk 10]
```diff
@@ -787,139 +776,212 @@ Storage Scenarios and Architecture Fit
...
-      1. Consuming and exposing Object storage through Tenant application
-      2. Use Embedded Shared File Storage for Control and Tenant Storage Needs
-
-   -  Specific Areas of Consideration:
-
-      1. Embedded Shared File Storage:
++----+--------------------+----+------------------------------------------------------------------------------------------------------------------------------------+
+| Use Case                | Description                                                                                                                             |
++====+====================+====+====================================================================================================================================+
+| Data-centre Storage     | Provide a highly reliable and scalable storage caapability taht has flexibilty to meet diverse needs                                    |
```
[the remark this answers is not in the document]
Comment (jbhartley) [reply doc/ref_model/chapters/chapter03.rst]: @pgoyal01  I believe that this section with incorrect spelling was removed when we (you) updated the table formatting.

Positive word on rst, is that it allows much better control of tables than we were able to achieve with "mark down". Though the are hard to "debug" when they have complex layout.

**Review by pgoyal01 — APPROVED** [review]

**Comment by wmk-admin** [comment]
Merging as in my view all requirements are met.  The contents is approved, and the line wrapping is OK.
If there are any other formatting issues, they can be addressed as PROOFREADING

**Pull request merged**